engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/RService.java

```diff
@@ -90,31 +90,38 @@ public Overseer analyze(AnalysisSyncRequestDTO analysis, File analysisDir, BiCon
                 return new FailedOverseer(Instant.now(), "Unable to resolve to [" + HOST_DOCKER_INTERNAL + "]", AnalysisRequestTypeDTO.R, e);
             }
         }
+
+        Path path = analysisDir.toPath();
+        AuthEffects auth = credentialProviders.stream().map(provider -> provider.apply(dataSource, path, path.toAbsolutePath().toString())).findFirst().orElse(null);
```

Comment:
Should we filter out null values from the `provider.apply`? If I am not mistaken, in case the provider does not fit, it returns null.

Reply:
Good catch